submissions/zalevskyi/friends-app/app.js

```diff
@@ -0,0 +1,133 @@
+const USERS_DATA = []
+const USERS_QUANTITY = 30
+const RESULTS = document.getElementById('results')
+const SORT = document.getElementById('sort')
+const FILTER = document.getElementById('filter')
+const REQUEST_PARAMETERS = {order: 'ascending'}
+let connectionAttemps = 1
+
+function connectToServer() {
+  RESULTS.innerHTML = `<div class='connection_message'>Please wait. Connecting to server. Attemp: ${connectionAttemps}</div>`
+  connectionAttemps++
+  fetchUserData(showResults, connectToServer)
+}
+function fetchUserData(onReady, onError) {
+  let apiUrl = 'https://randomuser.me/api/1.3/'
+  let fields = ['name', 'gender', 'location', 'dob', 'phone', 'picture']
+  fetch(`${apiUrl}?results=${USERS_QUANTITY}&inc=${fields.join(',')}`, {headers: {'Content-Type': 'application/json'}})
+    .then(response => {
+      if (!response.ok) {
+        throw Error(response.statusText)}
+      return response})
+    .then(response => response.json())
+    .then(responseJSON => {
+      if (responseJSON.hasOwnProperty('results')) {
+        pushUserData(responseJSON.results)
+        onReady()}
+      else onError()
+    }).catch(error => onError(error))
+}
+function pushUserData(randomUserResults) {
+  randomUserResults.forEach(user => USERS_DATA.push(
+      {name: user.name.first,
+       surname: user.name.last,
+       gender: user.gender,
+       age: user.dob.age,
+       country: user.location.country,
+       city: user.location.city,
+       phone: user.phone,
+       image128px: user.picture.large}))
+}
+function showResults() {
+  RESULTS.innerHTML = getUserDataToShow(USERS_DATA, REQUEST_PARAMETERS).map(user => getUserCard(user)).join('')
+  SORT.innerHTML = [`<div class='sort_line'>Sort by: ${getSortLinks(REQUEST_PARAMETERS.sort)}</div>`,
+                  `<div class='sort_line'> Order: ${getSortOrderLinks(REQUEST_PARAMETERS.order)}</div>`].join('')
+}
+function getUserDataToShow(userData, parameters) {
+  return sort(filter(userData, parameters), parameters)
+}
+function sort(userData, parameters) {
+  if (parameters.hasOwnProperty('sort')) {
+    let sortedData = sortAscending(userData, parameters.sort)
+    if (parameters.hasOwnProperty('order') && parameters['order']==='descending') sortedData.reverse()
```

Comment:
Please, use curly brackets everywhere. It is not consistent when you use them in some place and in some place don't.

Also, one time you can forget to place it and you will have tricky bug ;)